app/src/main/java/com/sopt/instagram/ui/story/member/MemberStoryFragment.kt

```diff
@@ -103,9 +86,31 @@ class MemberStoryFragment :
         }
     }
 
+    private fun setProgressBar() {
+        for (i in 1..viewModel.storyList.size) {
+            progressBars.add(
+                layoutInflater.inflate(
+                    R.layout.view_story_progress_bar,
+                    binding.layoutStoryProgressBar,
+                ),
+            )
+        }
+        setProgressBarBackground(0, R.drawable.shape_gray5_fill_1000_rect)
+    }
+
+    private fun setProgressBarBackground(childIndex: Int, backgroundDrawable: Int) {
+        binding.layoutStoryProgressBar.getChildAt(childIndex).background =
+            AppCompatResources.getDrawable(
+                requireContext(),
+                backgroundDrawable,
+            )
+    }
+
     private fun initStoryTagBtnClickListener() {
         binding.btnStoryTag.setOnSingleClickListener {
-            TagDialogFragment().show(parentFragmentManager, this.tag)
+            viewModel.currentStory.value?.id?.let { id ->
```

Comment:
let 함수 사용을 지양해야된다고 들었는데 아닌가요?
```suggestion
            if(viewModel.currentStory.value?.id != null){ id ->
            ~~~
            }
```

Reply:
let도 상황에 따라 유용하게 사용하는 경우가 존재하는 것 같습니다! 지금과 같이 [Mutable 변수의 null을 체크할 경우 let을 사용하면 scope 내부에서 immutable을 보장해 주어](https://tourspace.tistory.com/208) 유용하다고 하네요
그리고 아래처럼 `!= null`을 활용하거나 다른 방법으로 처리하면 코드가 너무 길어지더라구요...🤣
```
if (viewModel.currentStory.value?.id != null) {
                TagDialogFragment.newInstance(viewModel.currentStory.value?.id!!).show(parentFragmentManager, TAG_DIALOG)
            }
```

Reply:
새로운 정보를 얻어갑니다!